pkgs/native_assets_cli/lib/src/code_assets/validation.dart

```diff
@@ -198,7 +198,7 @@ void _validateCodeAssets(
   }
 
   final file = codeAsset.file;
-  if (file == null && !dryRun) {
+  if (file == null && !dryRun && _mustHaveFile(codeAsset.linkMode)) {
```

Comment:
The fix is maybe 10 lines of code. All the other things in here are 350 lines of code. I think this is a huge burden to maintain. Why can't we have one simple package, that emits various kids of code assets and executes them at runtime? Then the test in this PR would be maybe 30 lines of code (10 for a hook, 10 for native declarations and 10 for invoking and checking) instead of 10x this. The tests would also run much faster. Any good argument that speaks against this?

Reply:
350 lines:

* Example to illustrate how this works, to cover a user journey for Mariam.
* Integration test combines all the link modes that are not the bundling mode.

> and executes them at runtime

Yeah, that's not going to work until the fix rolls in to Dart and Flutter, so that's why `test_data/` is exercised in `dartdev`. I'll make sure we exercise it in Dartdev on the roll.

> I think this is a huge burden to maintain.

Objection noted. Agree to disagree with refactoring tools at our disposal.